aws/resource_aws_apigatewayv2_api.go

```diff
@@ -135,6 +141,39 @@ func resourceAwsApiGatewayV2Api() *schema.Resource {
 	}
 }
 
+func resourceAwsAPIGatewayV2ImportOpenAPI(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigatewayv2conn
+
+	if body, ok := d.GetOk("body"); ok {
+		revertReq := &apigatewayv2.UpdateApiInput{
+			ApiId:       aws.String(d.Id()),
+			Name:        aws.String(d.Get("name").(string)),
+			Description: aws.String(d.Get("description").(string)),
+			Version:     aws.String(d.Get("version").(string)),
+		}
+
+		log.Printf("[DEBUG] Updating API Gateway from OpenAPI spec %s", d.Id())
+		importReq := &apigatewayv2.ReimportApiInput{
+			ApiId: aws.String(d.Id()),
+			Body:  aws.String(body.(string)),
+		}
+
+		_, err := conn.ReimportApi(importReq)
+
+		if err != nil {
+			return fmt.Errorf("error importing API Gateway specification: %s", err)
+		}
+
+		log.Printf("[DEBUG] Reverting API Gateway v2 API: %s", revertReq)
+		_, err = conn.UpdateApi(revertReq)
+		if err != nil {
+			return fmt.Errorf("error reverting API Gateway v2 API (%s): %s", d.Id(), err)
```

Comment:
```suggestion
			return fmt.Errorf("error updating API Gateway v2 API (%s): %s", d.Id(), err)
```

It's good to have `reverting` in the log statement but would be confusing in the error message.